ouroboros-consensus/src/Ouroboros/Storage/FS/Sim/MockFS.hs

```diff
@@ -398,6 +398,14 @@ checkFsTree' ErrorHandling{..} = go
           }
     go (Left (FsMissing fp _)) =
         return (Left fp)
+    go (Left (FsExists fp)) =
+        throwError FsError {
+            fsErrorType   = FsResourceInappropriateType
```

Comment:
Surprised that "inapproproate type" is the right error here. If it is, we might want to document that as part of `open` (though if get rid of "inappropriate type" altogether perhaps we can give a somewhat more meaningful error here).

Reply:
It should be `FsResourceAlreadyExist`.